### DISCUSSION TITLE
Granular control of chats

### DISCUSSION THREAD

**Discussion by Keiron** [namumakwembo/wirechat#23] [Ideas]
Hi,

I noticed that it's possible to restrict who can create chats or groups, but I have a slightly different requirement.

Is there a way to restrict specific users so that they can only view certain groups and control whom they can initiate chats with?

For example, could we set it up so a user can only see chats relevant to their access tier and is only allowed to send messages to an admin user?

Thanks!

**Comment by namumakwembo** [comment]
Hello,

Thank you for your question!  

Currently, users are already restricted to viewing only the chats or groups they are part of as participants.  so only conversations the are a part of will appear in the chat list 

For controlling who can initiate chats, you might want to look into customizing the user `search` functionality in the chat creation modal. Specifically, you can restrict the users returned by modifying the `searchChatables($query)` method in the provided trait. Here’s an example of how you can achieve this:

```php
/**
 * Search for users when creating a new chat or adding members to a group.
 * Customize the search logic to limit results, such as restricting to specific users or criteria.
 */
public function searchChatables(string $query): ?Collection
{
    $searchableFields = ['name'];
    return User::where(function ($queryBuilder) use ($searchableFields, $query) {
        foreach ($searchableFields as $field) {
            $queryBuilder->orWhere($field, 'LIKE', '%'.$query.'%');
        }
    })
    ->limit(20)
    ->get();
}
```


You can find more details in the [https://wirechat.namuio.com/docs/trait](https://wirechat.namuio.com/docs/trait).  

If this approach doesn’t fully meet your requirements, feel free to share more about your use case. Since Wirechat is still in beta, we’re actively adding new features, and your feedback could help shape future functionality!  

Let me know how it works for you.

**Comment by Keiron** [comment]
So in theory I could do the following....

1. Add users to a group programmatically that was appropriate for their membership tier to meet their cohort and turn off the new create Group Modal, so they can't create new groups.

2. Change the search (as you've detailed) to only find admin users to create new messages with.

I have a project coming up that could really use this any idea how long you'll be in beta?

**Comment by namumakwembo** [comment]
Hello,

Yes, Wirechat can already do exactly what you’ve described!

As a developer, you have full control to turn off features like creating new groups or private chats globally via the configuration or on a per-user basis, as mentioned earlier. Additionally, groups can be created, and members can be added programmatically, giving you the flexibility to manage access based on membership tiers or other criteria.

I recommend checking out the [documentation](https://wirechat.namuio.com/docs/trait.) for more details on customization and configuration options.

Regarding the beta phase, the plan is to transition out of beta once the package has undergone sufficient testing and users report no significant issues. The  beta phase has been incredibly helpful in identifying and fixing bugs and errors that weren’t apparent during development. we’re making consistent progress towards  a stable release. 

Let me know if you have further questions or need assistance with implementation!
